File: sentence_analysis.py
#######################################
# Sentence Analysis Script
# Developed in VSCode Style
#######################################

# Initialize counters
from itertools import tee


character_count = 0
word_count = 0
vowel_count = 0

# List of vowels
vowels = "aeiouAEIOU"

# Input the sentence
print("Enter a sentence ending with a period:")
sentence = input()

# Check if the input ends with a period
if not sentence.endswith("."):
    print("Error: Sentence must end with a period.")
else:
    # Process each character in the sentence
    for i, char in enumerate(sentence):
        # Increment character count for each character except the period
        if char != ".":
            character_count += 1

        # Count words: A word starts after a space or is the first word in the sentence
        if char == " " and i > 0 and sentence[i - 1] not in " .":
            word_count += 1

        # Count vowels
        if char in vowels:
            vowel_count += 1

    # Include the last word if the sentence is not empty or invalid
    if sentence[-2] not in " .":
        word_count += 1

    # Display the results
    print("\n==============================")
    print("Analysis Results:")
    print("==============================")
    print(f"Length of the sentence (number of characters, excluding the period): {character_count}")
    print(f"Number of words in the sentence: {word_count}")
    print(f"Number of vowels in the sentence: {vowel_count}")
    print("==============================\n")

File: sentence_analysis.js
function analyzeSentence(sentence) {
    let length = 0;  // Counter for the length of the sentence
    let wordCount = 0;  // Counter for the number of words
    let vowelCount = 0;  // Counter for the number of vowels
    const vowels = "aeiouAEIOU";  // String of vowel characters
    let inWord = false;  // Boolean variable to track whether we're in a word

    // Loop through the sentence character by character
    for (let i = 0; i < sentence.length; i++) {
        const char = sentence[i];
        length++;  // Increment length for every character

        if (char === '.') {
            break;  // End of sentence, stop processing
        }

        if (char === ' ') {
            if (inWord) {  // If we were in a word and encounter space, count the word
                wordCount++;
            }
            inWord = false;  // We're no longer in a word
        } else {
            inWord = true;  // We encountered a non-space, so we're in a word
            if (vowels.includes(char)) {  // Check if the character is a vowel
                vowelCount++;
            }
        }
    }

    // If we were in a word and reached the end of the sentence, count the last word
    if (inWord) {
        wordCount++;
    }

    return { length, wordCount, vowelCount };
}

// Example usage
const sentence = "Hello there, how are you today.";
const result = analyzeSentence(sentence);

console.log("Length of sentence:", result.length);
console.log("Number of words:", result.wordCount);
console.log("Number of vowels:", result.vowelCount);
